connectivity/drivers/lora/TARGET_STM32WL/STM32WL_LoRaRadio.cpp

```diff
@@ -0,0 +1,1561 @@
+        lock();
+        if (flags & SIG_INTERRUPT) {
+            irq_process_isr();
+        }
+        unlock();
+    }
+}
+#endif
+
+void STM32WL_LoRaRadio::error_handler(HAL_StatusTypeDef error)
+{
+  printf ("ERROR in STM32WL_loRaRadio.c : %d", error);
+  /* USER CODE BEGIN error_handler_Debug */
+  /* User can add his own implementation to report the HAL error return state */
+  while (1)
+  {
+  }
+  /* USER CODE END error_handler_Debug */
+}
+
+void STM32WL_LoRaRadio::irq_process_isr()
+{
+#ifdef MBED_CONF_RTOS_PRESENT
+    irq_thread.flags_set(SIG_INTERRUPT);
+#else
+    RadioIrqProcess();
+#endif
+}
+
+uint16_t STM32WL_LoRaRadio::get_irq_status(void)
+{
+    uint8_t status[2];
+
+    read_opmode_command((uint8_t) RADIO_GET_IRQSTATUS, status, 2);
+    return (status[0] << 8) | status[1];
+}
+
+void STM32WL_LoRaRadio::clear_irq_status(uint16_t irq)
+{
+    uint8_t buf[2];
+
+    buf[0] = (uint8_t)(((uint16_t) irq >> 8) & 0x00FF);
+    buf[1] = (uint8_t)((uint16_t) irq & 0x00FF);
+    write_opmode_command((uint8_t) RADIO_CLR_IRQSTATUS, buf, 2);
+}
+
+//TODO - Better do CAD here. CAD code is already part of the driver
+//       It needs to be hooked up to the stack (this API will need change
+//       and the stack will need changes too)
+bool STM32WL_LoRaRadio::perform_carrier_sense(radio_modems_t modem,
+                                             uint32_t freq,
+                                             int16_t rssi_threshold,
+                                             uint32_t max_carrier_sense_time)
+{
+    bool status = true;
+    int16_t rssi = 0;
+
+    set_modem(modem);
+    set_channel(freq);
+    _reception_mode = RECEPTION_MODE_OTHER;
+    _rx_timeout = 0x00000000;
+    receive();
+
+    // hold on a bit, radio turn-around time
+    ThisThread::sleep_for(1);
+
+    Timer elapsed_time;
+    elapsed_time.start();
+
+    // Perform carrier sense for maxCarrierSenseTime
+    while (elapsed_time.read_ms() < (int) max_carrier_sense_time) {
+        rssi = get_rssi();
+
+        if (rssi > rssi_threshold) {
+            status = false;
+            break;
+        }
+    }
+
+    sleep();
+    return status;
+}
+
+void STM32WL_LoRaRadio::start_cad(void)
+{
+    // TODO: CAD is more advanced in SX126X. We will need API change in LoRaRadio
+    //       for this to act properly
+}
+
+
+
+/**
+ * TODO: The purpose of this API is unclear.
+ *       Need to start an internal discussion.
+ */
+bool STM32WL_LoRaRadio::check_rf_frequency(uint32_t frequency)
+{
+    // Implement check. Currently all frequencies are supported ? What band ?
+    return true;
+}
+
+void STM32WL_LoRaRadio::set_tx_continuous_wave(uint32_t freq, int8_t power,
+                                              uint16_t time)
+{
+    // This is useless. We even removed the support from our MAC layer.
+}
+
+
+/**
+  * @brief This function handles SUBGHZ Radio Interrupt.
+  */
+static void SUBGHZ_Radio_IRQHandler(void)
+{
+  /* USER CODE BEGIN SUBGHZ_Radio_IRQn 0 */
```

Comment:
do we want to keep the user sections generated from MX. Can be kept if this will be useful for later maintenance and in case the code is copy-paster directly from MX

Reply:
Better to remove them, not useful in mbed-os env: done